torch/_inductor/codegen/triton.py

```diff
@@ -1877,7 +1875,7 @@ def load(self, name: str, index: sympy.Expr):
             ep = ", eviction_policy='evict_last'"
         elif not is_coalesced:
             ep = ", eviction_policy='evict_last'"
-        elif self.inside_reduction and self.range_trees[-1].is_loop:
+        else:
```

Comment:
Rather than doing this, we probably just want to directly mark everything as `evict_first` for persistent reductions and pointwise ops.

Now, what about foreach kernels and split scans?